mesonbuild/utils/universal.py

```diff
@@ -2289,16 +2289,21 @@ def pickle_load(filename: str, object_name: str, object_type: T.Type) -> T.Any:
     except (pickle.UnpicklingError, EOFError):
         raise MesonException(load_fail_msg)
     except (TypeError, ModuleNotFoundError, AttributeError):
+        build_dir = os.path.dirname(os.path.dirname(filename))
         raise MesonException(
             f"{object_name} file {filename!r} references functions or classes that don't "
             "exist. This probably means that it was generated with an old "
-            "version of meson.")
+            "version of meson. Try running from the source directory "
+            f'meson setup {build_dir} --wipe')
     if not isinstance(obj, object_type):
         raise MesonException(load_fail_msg)
     from ..coredata import version as coredata_version
     from ..coredata import major_versions_differ, MesonVersionMismatchException
-    if major_versions_differ(obj.version, coredata_version):
-        raise MesonVersionMismatchException(obj.version, coredata_version)
+    version = getattr(obj, 'version', None)
+    if version is None:
+        version = obj.environment.coredata.version
```

Comment:
Isn't this just:
```python
version = getattr(obj, 'version', obj.environment.coredata.version)
```

Reply:
That's an error if it doesn't have an environment attribute.

Coredata and installdata only have a version attribute. build data only has an environment attribute.

We could add a version attribute to `build.Build` but that doesn't help to detect the problem of people loading build directories created by versions of Meson before 0.64.1

Reply:
Oh, right, because it's evaluated eagerly. Sigh